framework/aster-frame/src/mm/copy_with_ex_handling.S

```diff
@@ -0,0 +1,71 @@
+/* SPDX-License-Identifier: MPL-2.0 */
+
+// Copies `size` bytes from `src` to `dst`. This function is work with exception handling
+// and can recover from page fault. The source and destination must not overlap.
+// 
+// Returns number of bytes that failed to copy.
+//
+// Ref: [https://github.com/torvalds/linux/blob/2ab79514109578fc4b6df90633d500cf281eb689/arch/x86/lib/copy_user_64.S]
+// 
+// FIXME: enable exception table to avoid the recover workaround based on r15.
+.text
+.global copy_with_ex_handling
+.code64
+copy_with_ex_handling: # (dst: *mut u8, src: *const u8, size: usize) -> usize
```

Comment:
As stated in `rep_movs_alternative` (of Linux), this function is slower then rep movs in machines with FSRM. From our practice (#839), this is super slow on our machine which have FSRM but without fast short rep movs.

I think that the `r15` workaround works with the implementation of #845 (which is reverted). So just assume that we have FSRM and use rep movsb as a default.